ui/app/src/modules/Content/Authoring/NewContentDialog.tsx

```diff
@@ -149,18 +149,17 @@ interface NewContentDialogBaseProps {
   site: string;
   previewItem: PreviewItem;
   compact: boolean;
-
-  onSaveLegacySuccess?(response): any;
-
-  onSaveSuccess?(response): any;
+  onSaveSuccess?: StandardAction;
```

Comment:
Let’s change the API of this dialog so that it accepts a onContentTypeSelected(contentType: ContentType) instead